src/is_intersection_empty.jl

```diff
@@ -117,9 +126,47 @@ function is_intersection_empty(H1::AbstractHyperrectangle{N},
     return (false, v)
 end
 
+# --- disjointness check for 1D intervals ---
 
-# --- AbstractSingleton ---
+"""
+    is_intersection_empty(I1::Interval{N}, I2::Interval{N},
+                          witness::Bool=false) where {N<:Real}
+
+Check whether two intervals do not intersect, and otherwise optionally
+compute a witness.
+
+### Input
+
+- `I1`      -- first interval
+- `I2`      -- second interval
+- `witness` -- (optional, default: `false`) compute a witness if activated
+
+### Output
+
+* If `witness` option is deactivated: `true` iff ``I1 ∩ I2 = ∅``
+* If `witness` option is activated:
+  * `(true, [])` iff ``I1 ∩ I2 = ∅``
+  * `(false, v)` iff ``I1 ∩ I2 ≠ ∅`` and ``v ∈ I1 ∩ I2``
 
+### Algorithm
+
+``I1 ∩ I2 ≠ ∅`` iff there is a gap between the left-most point of the second
+interval and the left-most point of the first interval, or vice-versa.
+
+A witness is computed by taking the maximum over the left-most points of each
+interval, which is guaranteed to belong to the intersection.
+"""
+function is_intersection_empty(I1::Interval{N}, I2::Interval{N},
+                               witness::Bool=false) where {N<:Real}
+    check = min(I2) > max(I1) || min(I1) > max(I2)
+    if !witness
+        return check
+    else
+        return (check, N[max(min(I1), min(I2))])
```

Comment:
The `N` here is not needed. I am not sure if it makes the code more or less efficient.

Reply:
True; the elements of this vector are already of type `N`. I ran it and it doesn't seem to affect runtime. I'll remove it though, since it is not needed anyways.